esp-hal/src/dma/pdma.rs

```diff
@@ -927,14 +921,14 @@ crate::impl_dma_eligible!([I2s1DmaChannel] I2S1 => I2s1);
 pub struct Dma<'d> {
     _inner: PeripheralRef<'d, crate::peripherals::DMA>,
     /// DMA channel for SPI2
-    pub spi2channel: Spi2DmaChannelCreator,
+    pub spi2channel: Channel<'d, Blocking, Spi2DmaChannel>,
```

Comment:
I haven't looked at the future PRs yet, but this looks like an opportunity to remove `Dma` completely, and instead move the channels to `Peripherals` (the channel driver can check if DMA subsystem is enabled and enable it if required). Let's not worry about this for now though.

Reply:
Good idea but it'll have to wait.

Reply:
You only mean to do this on the  PDMA side right? (Since that better matches reality of the DMA there)

Reply:
> You only mean to do this on the PDMA side right? (Since that better matches reality of the DMA there)

Could you explain what you think the issue might be with adding GDMA channels to `Peripherals`?

Reply:
From the perspective of a 3rd party driver, the invariant that having an instance of a DMA channel means the GDMA peripheral has been enabled is very useful. Especially when you start adding destructors to disable it. Playing the game of "enable it if it's not already enabled" is annoying and prone to race conditions.

Reply:
A 3rd party driver will construct a Channel/ChannelRx/ChannelTx structure internally (as we do in upcoming PRs) which will do the initialization for them. Unless they don't use it and deal with the problem on their own.

Reply:
The `Channel` structures unfortunately only expose a limited subset of the DMA channel's APIs. A common reason to write a 3rd party driver is because the hal driver doesn't expose some features. Using a `Channel` gets in the way.